pkg/apis/authorization/types.go

```diff
@@ -144,3 +144,53 @@ type SubjectAccessReviewStatus struct {
 	// For instance, RBAC can be missing a role, but enough roles are still present and bound to reason about the request.
 	EvaluationError string
 }
+
+// +genclient=true
+// +noMethods=true
+
+// SelfSubjectRulesReview enumerates the set of actions the current user can perform within a namespace.
+// The returned list of actions may be incomplete depending on the server's authorization mode,
+// and any errors experienced during the evaluation.
+type SelfSubjectRulesReview struct {
+	metav1.TypeMeta
```

Comment:
Without namespace information, the serialization of this type cannot stand alone.